db/migrate/20230605133804_remove_master_variants.rb

```diff
@@ -4,18 +4,19 @@ def change
       delete_master_option_values
     end
 
-    delete_master_line_items
+    handle_master_line_items
     delete_master_inventory_units
     delete_master_variant_prices
     delete_master_variants
   end
 
   private
 
-  def delete_master_line_items
+  def handle_master_line_items
     ActiveRecord::Base.connection.execute(<<-SQL
-      DELETE FROM spree_line_items
-      USING spree_variants
+      UPDATE spree_variants
+      SET is_master = false
+      FROM spree_line_items
```

Comment:
That's brilliant! If the variant is referenced then we should turn it into a real variant. :star: